File: languages/python/bloock/entity/encryption/encrypter_args.py
from bloock.entity.key.local_key import LocalKey
from bloock.entity.key.managed_key import ManagedKey


class EncrypterArgs:
    local_key = None
    managed_key = None

    def __init__(self, key) -> None:
        if isinstance(key, LocalKey):
            self.local_key = key
        elif isinstance(key, ManagedKey):
            self.managed_key = key
        else:
            raise Exception("Invalid key provided. Must be of type LocalKey or ManagedKey")

File: languages/php/Bloock/Entity/Encryption/EncrypterArgs.php
<?php

namespace Bloock\Entity\Encryption;

use Bloock\Entity\Key\LocalKey;
use Bloock\Entity\Key\ManagedKey;
use Exception;

class EncrypterArgs
{
    public ?LocalKey $localKey = null;
    public ?ManagedKey $managedKey = null;

    /**
     * @throws Exception
     */
    public function __construct($key)
    {
        if ($key instanceof LocalKey) {
            $this->localKey = $key;
        } else if ($key instanceof ManagedKey) {
            $this->managedKey = $key;
        } else {
            throw new Exception("Invalid $key provided. Must be of type LocalKey or ManagedKey");
        }
    }
}